src/Connection.php

```diff
@@ -526,9 +526,13 @@ public function setAutoCommit($autoCommit)
      *
      * @throws Exception
      */
-    public function fetchAssociative(string $query, array $params = [], array $types = [])
-    {
-        return $this->executeQuery($query, $params, $types)->fetchAssociative();
+    public function fetchAssociative(
+        string $query,
+        array $params = [],
+        array $types = [],
+        ?QueryCacheProfile $qcp = null
```

Comment:
Is this really necessary? There are already too many optional parameters in these methods.

The query builder should be able to call `$connection->executeQuery(..., $qcp)-> fetchAssociative()` instead.

Reply:
You are right it is not necessary at all for `Connection` class, I use `QueryBuilder` 100% time I will update the code soon. Thanks for reply and good point.

Reply:
I have updated the code and now I will think about how to test it, whether to extend the existing fetch* tests or create a new set testing methods for caching. What do you think?